(WIP) Feature/6224 support

jovial opened the pull request.

jovial committed:
Add privilege exec configuration option

This inputs the command `enable` before attempting
any config. This is necessary to support the Dell
powerconnect 6224 line of switches.

jovial committed:
Add dell_powerconnect_port_type config option

This controls <port type> in commands of the
form: interface <port type> <port>

This is needed to support the Dell PowerConnect
6224 line of switches

markgoddard commented on defaults/main.yml:
```diff
@@ -9,3 +9,13 @@ dell_powerconnect_switch_config: []
+# This is required if your switch requires you to enter privilege
+# exec mode to perform switch configuration
+# FIXME: password prompt not currently supported
+dell_powerconnect_enable_privilege_exec_mode: false
```
Does this need to be conditional? If the switch does not use enable mode we'll just not get the enable prompt

Commit:
Fix some issues with dell powerconnect 6224

* Look for invalid command message
* Login to SSH for interface configuration
* Make regexes more robust

jovial committed:
Rename port_type to interface_type

This more accurately desribes the variable, as port type is
a term used in the configuration help:

ethsw-dell-6224-1(config)#interface ethernet ?

<unit>/<port-type><port> Enter interface in interface format.

jovial committed:
update readme to add interface_type option

jovial committed:
Remove dell_powerconnect_switch_enable_privilege_exec flag

Existing behavior shouldn't be affected ans this simplifies
use of this role.

jovial committed:
Revert "update readme to add interface_type option"

This reverts commit 1b6b02f32ff9fb224a1ecc68ccf0b5aac13d370a.

jovial committed:
Remove interface_type option

It makes sense to include this in the interface description,
e.g 'ethernet 1/g1` so that you can configure multiple interfaces.
